graphbolt/src/csc_sampling_graph.cc

```diff
@@ -115,5 +120,46 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::InSubgraph(
           : torch::nullopt);
 }
 
+c10::intrusive_ptr<CSCSamplingGraph> CSCSamplingGraph::CopyToSharedMemory(
+    const std::string& shared_memory_name) {
+  auto shm_name_with_prefix = std::string("graphbolt_") + shared_memory_name;
+  std::vector<torch::Tensor> tensors = {indptr_, indices_};
+  if (node_type_offset_.has_value()) {
+    tensors.push_back(node_type_offset_.value());
+    tensors.push_back(type_per_edge_.value());
+  }
+  SharedMemoryPtr tensor_meta_shm, tensor_data_shm;
+  std::vector<torch::Tensor> shared_tensors;
+  std::tie(tensor_meta_shm, tensor_data_shm, shared_tensors) =
+      CopyTensorsToSharedMemory(
+          shm_name_with_prefix, tensors, SERIALIZED_METAINFO_SIZE_MAX);
+  return c10::make_intrusive<CSCSamplingGraph>(
+      shared_tensors[0], shared_tensors[1],
+      shared_tensors.size() > 2
+          ? torch::optional<torch::Tensor>(shared_tensors[2])
+          : torch::optional<torch::Tensor>(),
+      shared_tensors.size() > 3
+          ? torch::optional<torch::Tensor>(shared_tensors[3])
+          : torch::optional<torch::Tensor>(),
+      std::move(tensor_meta_shm), std::move(tensor_data_shm));
```

Comment:
After the renaming in graphbolt/src/shared_memory_utils.h, I would create a helper function in this cc file to covert SharedMemoryTensors to c10::make_intrusive<CSCSamplingGraph>.